pkg/apis/pipeline/v1beta1/resolver_types.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// ResolverRef can be used to refer to a Pipeline or Task in a remote
+// location like a git repo. This feature is in alpha and these fields
+// are only available when the alpha feature gate is enabled.
+type ResolverRef struct {
+	// Resolver is the name of the resolver that should perform
+	// resolution of the referenced Tekton resource, such as "git".
+	// +optional
+	Resolver string `json:"resolver,omitempty"`
```

Comment:
The TEP [proposes](https://github.com/tektoncd/community/blob/main/teps/0060-remote-resource-resolution.md#resolver-specifics) a specific set of resolvers: `git`, `bundle` and `in-cluster`  

What do you think about using aliases for the strings, instead of strings directly, and adding validation for resolvers themselves? 

K8s API convention encouraging use of string aliases when fields will have a list of allowed values (enumerations): https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants

Reply:
This field is intentionally free-form. The resolution feature is intended to allow integration with any resolver an operator wants to install and that could include ones that Tekton hasn't explicitly named in its source code. e.g. "svn", "bzr", "gcs"

So, I guess I don't see this being in the spirit of what the Resolution TEP is aiming to achieve for the Tekton project but I also don't want to introduce freeform data if you feel strongly that you'd prefer to have the options explicitly named. Let me know.

Reply:
Thanks for the additional context!

How about we declare the `Resolver` type, specify the ones we provide, then integrators can specify their own:

```go
# in tekton pipelines
type Resolver string

const (
	git     Resolver = "git"
	bundle  Resolver = "bundle"
	cluster Resolver = "cluster"
)
```

```go
# in an integration
const (
	svn tekton.Resolver = "svn"
	bzr tekton.Resolver = "bzr"
	gcs tekton.Resolver = "gcs"
)
```

Similarly to how we use [selection.Operator](https://github.com/tektoncd/pipeline/blob/ecb251c18071deb241ed57e6826c252ce40b22bf/vendor/k8s.io/apimachinery/pkg/selection/operator.go#L21) from K8s apimachinery in [operator](https://github.com/tektoncd/pipeline/blob/adf1953542467da17e189f3fb7b7b1a3a3a2050c/pkg/apis/pipeline/v1beta1/when_types.go#L33) field in `when` expressions 

What do you think?

Reply:
OK, I've updated the type from `string` to `v1beta1.ResolverName`.